### PR TITLE
Added package to compile and link yaml parser in C++

### PR DESCRIPTION
**Pull request by abdosi**
What I did:
Added package to compile and link yaml file parser in C++

Why I did:

To compile and link changes done in PR: https://github.com/sonic-net/sonic-swss/pull/3353

How I verify:
Manual Verification

### CODE REVIEW

**Comment by abdosi** [comment]
@saiarcot895 : can you please help in review this.

File: dockers/docker-fpm-frr/Dockerfile.j2 [hunk 1]
```diff
@@ -16,6 +16,7 @@ RUN apt-get update   && \
         libc-ares2      \
         iproute2        \
         logrotate       \
+        libyaml-cpp-dev \
```
Comment (saiarcot895) [inline dockers/docker-fpm-frr/Dockerfile.j2:19]: Assuming you don't actually need the development headers at runtime, there shouldn't be any changes needed in this file. Dependency resolution should be able to install the runtime library automatically.

Comment (abdosi) [reply dockers/docker-fpm-frr/Dockerfile.j2:19]: @saiarcot895 : not sure if i completely follow this comment. Without this change I was getting run-time error when `fpmsyncd` was running in bgp docker. PR for reference: https://github.com/sonic-net/sonic-swss/pull/3353

**Comment by abdosi** [comment]
@saiarcot895 : can we get signoff on this,

File: dockers/docker-fpm-frr/Dockerfile.j2 (hunk 1 above)
Comment (abdosi) [reply dockers/docker-fpm-frr/Dockerfile.j2]: thanks @saiarcot895 , i have removed the change. Yes, as part of the build of `docker-swss-layer-bookworm.gz` all swss dependency are resolved and we don't need to explicit install on in frr docker.

**Review by saiarcot895 — APPROVED** [review]

**Comment by abdosi** [comment]
'/azpw ms_conflict'

**Comment by abdosi** [comment]
@rlhui : can you help merge this.

**Review by rlhui — APPROVED** [review]

**Comment by yejianquan** [comment]
Having conflict @abdosi , please create a seperate PR to 202405 and add to wishlist

**Comment by abdosi** [comment]
> Having conflict @abdosi , please create a seperate PR to 202405 and add to wishlist

@yejianquan : PR for 202405: https://github.com/sonic-net/sonic-buildimage/pull/20856

**Pull request merged**